packages/node_modules/@webex/webex-core/src/lib/services/service-catalogs.js

```diff
@@ -0,0 +1,156 @@
+import {SERVICE_CATALOGS} from './constants';
+import ServiceHost from './service-host';
+
+/**
+ * The parameter transfer object for {@link ServiceCatalogs#mapRemoteCatalog}.
+ * This object is shaped to match the object returned from the **U2C** service.
+ *
+ * @typedef {Record<string, string>} RSL
+ * @typedef {Record<string, Array<Record<string, number | string>>>} RHC
+ *
+ * @typedef {Object} MapRemoteCatalogPTO
+ * @property {string} MapRemoteCatalogPTO.catalog - Service catalog name.
+ * @property {RSL} MapRemoteCatalogPTO.serviceLinks - Service links.
+ * @property {RHC} MapRemoteCatalogPTO.hostCatalog - Service host catalog.
+ */
+
+/**
+ * @class
+ * @classdesc - Manages a collection of {@link ServiceHost} class objects.
+ */
+export default class ServiceCatalogs {
+  /**
+   * Generate a new {@link ServiceHost}.
+   *
+   * @public
+   * @constructor
+   * @memberof ServiceHost
+   */
+  constructor() {
+    /**
+     * The collection of managed {@link ServiceHost}s.
+     *
+     * @instance
+     * @type {Array<ServiceHost>}
+     * @private
+     * @memberof ServiceCatalogs
+     */
+    this.hosts = [];
+  }
+
+  /**
+   * Load a formatted array of {@link ServiceHost} constructor parameter
+   * transfer objects as instances of {@link ServiceHost} class objects to the
+   * {@link ServiceCatalogs#hosts} array.
+   *
+   * @public
+   * @memberof ServiceCatalogs
+   * @param {Array<ServiceHost.ConstructorPTO>} hosts
+   * @returns {this}
+   */
+  load(hosts = []) {
+    // Validate that the provided hosts are eligible to be loaded.
+    const validHosts = hosts.filter(
+      (host) => !!(ServiceCatalogs.mapCatalogName({
+        id: host.catalog,
+        type: 'string'
+      }))
+    );
+
+    // Load the eligible hosts.
+    this.hosts.push(...validHosts.map(
+      (loadableHost) => new ServiceHost(loadableHost)
+    ));
+
+    return this;
+  }
+
+  /**
+   * Convert a {@link SERVICE_CATALOGS} identifier or value to its associated
+   * idenfier or value.
+   *
+   * @public
+   * @static
+   * @memberof ServiceCatalogs
+   * @param {Object} pto - The parameter transfer object.
+   * @property {string | number} pto.id - The identifier to convert in the enum.
+   * @property {'string' | 'number'} pto.type - The desired output.
+   * @returns {string|number} - The matching enum value or index.
+   */
+  static mapCatalogName(pto) {
```

Comment:
desctructure since `pto` itself isn't used

Reply:
Addressed this, destructuring the parameter transfer object now.